pywwt/layers.py

```diff
@@ -276,49 +276,48 @@ def add_image_layer(
             Use this parameter to set a display name for the image set.
             If not set, pywwt will set a name based on the file name of the
             provided image.
-        force_hipsgen : optional boolean, defaults to False
-            Force the input FITS to be tiled with HiPSgen. If this and
-            *force_tan* are set to False, this method will figure out when to
-            split the input into tiles and also which type of projection to
-            use. Tangential projection for smaller angular areas and HiPSgen
-            larger regions of the sky.
-        force_tan : optional boolean, defaults to False
-            Force the input FITS to be tiled with a tangential projection. If
-            this and *force_hipsgen* are set to False, this method will figure
-            out when to split the input into tiles and also which type of
-            projection to use. Tangential projection for smaller angular areas
-            and HiPSgen larger regions of the sky.
+        tiling_method : optional :class:`~toasty.TilingMethod`
+            Can be used to force a specific tiling method, i.e. tiled
+            tangential projection, TOAST, HiPS, or even untiled. Defaults
+            to auto-detection, which choses the most appropriate method.
         kwargs
             Additional keyword arguments can be used to set properties on the
             image layer or settings for the `toasty` tiling process. Common
-            toasty settings include ``out_dir``, ``override``, and ``blankval``.
+            toasty settings include ``out_dir``, ``override``, ``blankval``,
+            and ``start``.
             See `toasty.tile_fits`.
 
         Returns
         -------
         layer : :class:`~pywwt.layers.ImageLayer` or a subclass thereof
         """
 
+        if isinstance(image, tuple):
+            image, wcs = image
+            from astropy.io.fits import PrimaryHDU
+
+            image = PrimaryHDU(image, wcs.to_header())
         if (
             isinstance(image, astropy.io.fits.ImageHDU)
             or isinstance(image, astropy.io.fits.PrimaryHDU)
             or isinstance(image, astropy.io.fits.HDUList)
         ):
-            unused_file_name = self._get_unused_fits_name()
-            image.writeto(unused_file_name)
-            image = unused_file_name
+            filename = "fits_input_{}".format(
+                hash(str(image.data.tobytes()) + image.header.tostring())
+            )
```

Comment:
This bit of code reads in the *entire* image data array, which could be very large. For instance, in my DASCH test image, the `tobytes()` representation is about 3 gigabytes. We should probably limit this to the first 64k of data, or something like that.

Calling `str` on the `tobytes()` return value is also inefficient. In my sample, it returns a string that starts `b'\\x7f\\xf8\\x00\\x00...`. The string is 10 gigabytes!

Finally, I believe that the Python built-in `hash()` function now has some randomization built in so that hash values vary from one Python invocation to the next. So the filenames won't be stable across runs or Jupyter kernels, which seems to go against the intent of this code. This can be dealt with by using a stably-defined hash, such as:

```python
from hashlib import md5
d = md5()
d.update(image.data[:65536])
d.update(image.header.tostring().encode('utf-8'))
h = d.hexdigest()  # returns digest as a hex-encoded string
```